### ISSUE TITLE
Support updating configmap for local-storage provisioner

### ISSUE THREAD

**Issue by gnufied** [kubernetes-incubator/external-storage#1011]
Currently when user edits the configmap that holds local storage configuration, all pods managed by daemonset must be deleted and recreated. This is rather awkward I think because the only way to do so is by using a selector and deleting created pods. @jsafrane also pointed out, it potentially could be problematic to delete local storage provisioner pod if it is in middle of an operation.

I have been wondering if provisioner itself should support automatically reloading the configmap so as new PVs can be created without having to delete and recreate the pods.

cc @msau42 @lichuqiang

**Comment by lichuqiang** [comment]
Good point! A file watcher to support automatically reloading is more friendly to users.
But as @msau42 mentioned, we need to investigate existing configs to avoid potential corner cases. say the mode in which to delete a pv, if its storage class has been removed from configuration.

And a small reminder of the filewatcher lib: I remembered that the filewatcher can not work well on symlink files, we might need extra logic to make sure it works on k8s configmap (maybe watch its parent path).
